backend/src/bundles/users/user.service.ts

```diff
@@ -27,20 +39,37 @@ class UserService implements IService {
         };
     }
 
-    public async create(
-        payload: UserSignUpRequestDto,
-    ): Promise<UserSignUpResponseDto> {
+    public async create({
+        email,
+        password,
+        firstName,
+        lastName,
+    }: UserSignUpRequestDto): Promise<Pick<UserEntityFields, 'id'>> {
+        const { hash: passwordHash, salt: passwordSalt } =
+            encryptPassword(password);
+        const { id: profileId } = await this.profileRepository.create({
+            firstName,
+            lastName,
+        });
+
         const user = await this.userRepository.create(
```

Comment:
If this fails, the profile will remain in the database. Saving to the database must be done in one request

Reply:
Good point, thanks for noticing, added the transaction.